### PR TITLE
allow special names to be uses as field name

### LINKED ISSUE #1523: Parse errors in SDL field definitions when using keywords as field names
**Describe the bug**
Given the following SDL:

```graphql
type Foo {
  true: String
  false: String
  on: String
  null: String
}
```

The parser throws a parse error even though these names are allowed as field names in the GraphQL grammar: https://graphql.github.io/graphql-spec/June2018/#FieldDefinition

I see there's already an attempt at handling this for other keywords in https://github.com/graphql-java/graphql-java/blob/fccc0c754eea96c6b89024349f359d2e563e6691/src/main/antlr/GraphqlCommon.g4#L31

**To Reproduce**
New up the GraphQL Parser and parse the example above:

```java
import graphql.parser.Parser;
new Parser().parseDocument(exampleSDLAsString);
```
Observe that the parser throws errors even though the above is valid according to the grammar.

Related issue: https://github.com/jimkyndemeyer/js-graphql-intellij-plugin/issues/239

### PR DESCRIPTION
**Pull request by andimarek**
fixes also enum value names and fragment names

this fixed #1523

### CODE REVIEW

File: src/main/java/graphql/parser/Parser.java [hunk 1]
```diff
@@ -18,6 +18,11 @@
 @Internal
 public class Parser {
 
+
+    public static Document parse(String input) {
+        return new Parser().parseDocument(input);
+    }
```
Comment (bbakerman) [inline src/main/java/graphql/parser/Parser.java]: yah

**Comment by PowerKiKi** [comment]
This bug fix has been merged almost two months ago, but was not released yet. Is there any chance to have a new release in the foreseeable future ?

The bug is rather annoying as it leads to critical failure as described in https://github.com/jimkyndemeyer/js-graphql-intellij-plugin/issues/239 and https://github.com/jimkyndemeyer/js-graphql-intellij-plugin/issues/264, so it would be very appreciated if the situation could be unblocked.

**Comment by smeevil** [comment]
I'm also eagerly waiting for this. Is there any window of opportunity of pushing this through ?

**Comment by PowerKiKi** [comment]
According to https://www.graphql-java.com/blog/breaking-changes-and-lts/:

> We are releasing new major versions of GraphQL Java roughly every 2 months. 

Unfortunately [last major release was on June 13](https://github.com/graphql-java/graphql-java/releases/tag/v13.0). So v14 is more than 3 months late. Seeing [the milestone](https://github.com/graphql-java/graphql-java/milestone/25), I can only guess that those 9 issues are what block a release.

I would suggest to move those 9 issues to v15 and release a v14 now. Because v14 already address 52 issues. That's certainly worth releasing :heart:

**Pull request merged**